lib/clientv3/ippool_kdd_conversion_test.go

```diff
@@ -0,0 +1,249 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv3_test
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/projectcalico/libcalico-go/lib/apiconfig"
+	apiv1 "github.com/projectcalico/libcalico-go/lib/apis/v1"
+	apiv3 "github.com/projectcalico/libcalico-go/lib/apis/v3"
+	"github.com/projectcalico/libcalico-go/lib/backend"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/clientv3"
+	"github.com/projectcalico/libcalico-go/lib/ipip"
+	"github.com/projectcalico/libcalico-go/lib/options"
+	"github.com/projectcalico/libcalico-go/lib/testutils"
+	"github.com/projectcalico/libcalico-go/lib/watch"
+)
+
+var _ = testutils.E2eDatastoreDescribe("IPPool KDD v1 to v3 migration tests", testutils.DatastoreK8s, func(config apiconfig.CalicoAPIConfig) {
+
+	ctx := context.Background()
+	name1 := "ippool-1"
+	name2 := "ippool-2"
+	spec1_v1 := apiv3.IPPoolSpec{
+		CIDR:        "1.2.3.0/24",
+		NATOutgoing: true,
+		IPIP: &apiv1.IPIPConfiguration{
+			Enabled: true,
+			Mode:    ipip.CrossSubnet,
+		},
+	}
+	spec1_v3 := apiv3.IPPoolSpec{
+		CIDR:        "1.2.3.0/24",
+		NATOutgoing: true,
+		IPIPMode:    apiv3.IPIPModeCrossSubnet,
+	}
+	kvp1_v3 := &model.KVPair{
```

Comment:
Is this really v3?

Reply:
I'm calling it kvp_v3 because it's the v3 style KVPair with the whole resource being the Value, but I'm happy to change it to just `kvp`